includes/Model/Story.php

```diff
@@ -133,8 +133,8 @@ public function load_from_post( $_post ) {
 			return false;
 		}
 
-		$this->title   = get_the_title( $post );
-		$this->excerpt = get_the_excerpt( $post );
+		$this->title   = $post->post_title;
+		$this->excerpt = $post->post_excerpt;
```

Comment:
Why was this changed?

Reply:
on web-stories archive pages we run few filters specifically `the_excerpt` and `the_content`, in this filters we try to embed the player. 
https://github.com/rtCamp/web-stories-wp/blob/49b01cea256adf5367155bcf84d5f430c4a78bf7/includes/Story_Post_Type.php#L714

Now here we are using `$story->load_from_post( $post );` which in turn used `get_the_excerpt()`. And this technically shouldn't collide. but on the archive page due to this it was going in too deep and resulting in page not loading and causing 502 server error.

Using data already fetched above with `get_post()` resolved the issue.

Reply:
Got it! Bit of a recursion there.

Change makes sense then.

I'll need to think about improving the archive page anyway 🤔

Reply:
we can now make the archive page use grid view. with Lightbox, but if we need individual click through for the single stories then may be disable the lightbox for archive page. 🤔

Reply:
Archive pages are styled by the theme now, so can't just make a grid.

But we have an existing ticket for allowing user customization.

Anyway, not a concern right now :-)